File: src/main/java/me/wonwoo/domain/repository/BlogHistoryRepository.java
package me.wonwoo.domain.repository;

import me.wonwoo.domain.model.BlogHistory;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.elasticsearch.repository.ElasticsearchCrudRepository;

import java.time.LocalDateTime;

/**
 * Created by wonwoo on 2017. 2. 14..
 */
public interface BlogHistoryRepository extends ElasticsearchCrudRepository<BlogHistory, Long> {
  Page<BlogHistory> findByDateBetween(LocalDateTime start, LocalDateTime end, Pageable pageable);
}

File: src/main/java/me/wonwoo/domain/repository/BlogHistoryRepository.scala
package me.wonwoo.domain.repository

import java.awt.print.Pageable
import java.time.LocalDateTime

import me.wonwoo.domain.model.BlogHistory
import org.springframework.data.elasticsearch.repository.ElasticsearchCrudRepository

/**
  * Created by wonwoo on 2016. 9. 21..
  */
trait BlogHistoryRepository extends ElasticsearchCrudRepository[BlogHistory, java.lang.Long] {

  def findByDateBetween(start: LocalDateTime, end: LocalDateTime, pageable: Pageable);
}
